src/main/java/seedu/smarthomebot/logic/commands/OnCommand.java

```diff
@@ -14,55 +17,74 @@
-    private final String key;
+    private final String argument;
     private final String parameter;
 
-    public OnCommand(String key, String parameter) {
-        this.key = key;
+    public OnCommand(String argument, String parameter) {
+        this.argument = argument;
         this.parameter = parameter;
     }
 
     @Override
     public CommandResult execute() {
-        String onByType = APPLIANCE_TYPE;
-        // To check if any of the appliances contains the name of the location
-        ArrayList<Appliance> filterApplianceList =
-                (ArrayList<Appliance>) applianceList.getAllAppliance().stream()
-                        .filter((s) -> s.getLocation().equals(this.key))
-                        .collect(toList());
-
-        // if list is empty
-        if (!filterApplianceList.isEmpty()) {
-            onByType = LOCATION_TYPE;
-        }
-        switch (onByType) {
-        case (APPLIANCE_TYPE):
-            return onByAppliance();
-        case (LOCATION_TYPE):
-            return onByLocation();
-        default:
-            return new CommandResult("");
+        try {
+            String onByType = APPLIANCE_TYPE;
+            ArrayList<Appliance> filterApplianceList =
+                    (ArrayList<Appliance>) applianceList.getAllAppliance().stream()
+                            .filter((s) -> s.getLocation().equals(argument))
+                            .collect(toList());
+
+            if (!filterApplianceList.isEmpty()) {
+                onByType = LOCATION_TYPE;
+            }
+            switch (onByType) {
+            case (APPLIANCE_TYPE):
+                return onByAppliance();
+            case (LOCATION_TYPE):
+                return onByLocation();
+            default:
+                return new CommandResult("Invalid Format");
+            }
+        } catch (ApplianceNotFoundException e) {
+            return new CommandResult(MESSAGE_APPLIANCE_OR_LOCATION_NOT_EXIST);
+        } catch (NoParameterForLocationException e) {
+            return new CommandResult(MESSAGE_NO_PARAMETER_IN_ON_BY_LOCATION);
+        } catch (NoApplianceInLocationException e) {
+            return new CommandResult("There is no appliance in \"" + argument + "\".");
         }
     }
 
-    private int getApplianceToOnIndex() {
+
+    private CommandResult onByAppliance() throws ApplianceNotFoundException, NoApplianceInLocationException {
+        int toOnApplianceIndex = getApplianceToOnIndex();
+        Appliance toOnAppliance = applianceList.getAppliance(toOnApplianceIndex);
+        String outputResult = onAppliance(toOnAppliance, "", false);
+        return new CommandResult(outputResult);
+    }
+
+    private int getApplianceToOnIndex() throws ApplianceNotFoundException, NoApplianceInLocationException {
         for (Appliance appliance : applianceList.getAllAppliance()) {
-            if (appliance.getName().equals((this.key))) {
+            if (appliance.getName().equals((argument))) {
```

Comment:
Should we use this.argument to indicate that we are accessing the current's object's variable? Other classes are using 'this' as well

Reply:
As discussed, we dont need the this .